Include/pymacro.h

```diff
@@ -118,6 +118,8 @@
         "We've reached an unreachable state. Anything is possible.\n" \
         "The limits were in our heads all along. Follow your dreams.\n" \
         "https://xkcd.com/2200")
+#elif defined(__GNUC__) || defined(__clang__)
```

Comment:
For MSVC `defined(_MSC_VER)` you can use `__assume(0)`: https://docs.microsoft.com/en-us/cpp/intrinsics/assume?view=vs-2019